substrate/frame/revive/src/limits.rs

```diff
@@ -109,6 +108,13 @@ pub mod code {
 	/// The code is stored multiple times as part of the compiled program.
 	const EXTRA_OVERHEAD_PER_CODE_BYTE: u32 = 4;
 
+	/// The maximum size of a basic block in number of instructions.
+	///
+	/// We need to limit the size of basic blocks because the interpreters lazy compilation
+	/// compiles one basic block at a time. A malicious program could trigger the compilation
+	/// of the whole program by creating one giant basic block otherwise.
+	const BASIC_BLOCK_SIZE: u32 = 200;
```

Comment:
200 is probably too tight: [this sha1 implementation](https://github.com/paritytech/revive/blob/main/crates/integration/contracts/SHA1.sol) peaks with a 584 instructions BB and there is nothing crazy inside that contract. BBs could get even larger than that.

Reply:
I assume because 256bit arithmetic will generate a lot of instructions on 32bit. Upped to `1000`. I like round numbers.